packages/itwin/imodel-react-hooks/src/utils/basic-hooks.ts

```diff
@@ -25,19 +25,19 @@ export function useOnMountInRenderOrder(effect: () => void | (() => void)) {
 
 /** run code on component mount */
 export function useOnMount(impl: () => void) {
-  useEffect(() => void impl(), []);
+  useEffect(() => void impl(), [impl]);
 }
 
 /** run something on component unmount */
 export function useOnUnmount(impl: () => void) {
-  useEffect(() => () => impl(), []);
+  useEffect(() => () => impl(), [impl]);
 }
 
 /** create a stable object reference on intialization */
 export function useStable<T>(make: () => T) {
   // can't use memo since it's only for expensive computations, react will (eventually)
   // selectively recreate the object breaking the contract.
-  const value = useMemo(make, []);
+  const value = useMemo(make, [make]);
```

Comment:
probably can replace `useMemo` with another ref and guarantee one call to the expensive function without any dependencies, like so:
```tsx
const valueRef = useRef<T>(undefined as any);
const hasMade = useRef(false);
if (!hasMade) {
  valueRef.current = make();
  hasMade.current = true;
}
return valueRef.current;
```

Reply:
oh yeah. Figures considering I didn't actually try my code out